diff --git a/node_modules/@typechain/ethers-v5/dist/codegen/index.js b/node_modules/@typechain/ethers-v5/dist/codegen/index.js
index 94e34bd..a12d615 100644
--- a/node_modules/@typechain/ethers-v5/dist/codegen/index.js
+++ b/node_modules/@typechain/ethers-v5/dist/codegen/index.js
@@ -137,8 +137,16 @@ function codegenContractFactory(codegenConfig, contract, abi, bytecode) {
 
   export class ${contract.name}${common_1.FACTORY_POSTFIX} extends ContractFactory {
     ${generateFactoryConstructor(codegenConfig, contract, bytecode)}
-    override deploy(${constructorArgs}): Promise<${contract.name}> {
-      return super.deploy(${constructorArgNames}) as Promise<${contract.name}>;
+    private static _deployed: Contract;
+    static get deployed(): ${contract.name} {
+     return this._deployed as ${contract.name};
+    }
+    static setAsDeployed(contract: Contract): ${contract.name}{
+     this._deployed = contract;
+     return contract as ${contract.name};
+    }
+    async deploy(${constructorArgs}): Promise<${contract.name}> {
+      return ${contract.name}${common_1.FACTORY_POSTFIX}.setAsDeployed(await super.deploy(${constructorArgNames}));
     }    
     override getDeployTransaction(${constructorArgs}): TransactionRequest {
       return super.getDeployTransaction(${constructorArgNames});
